controllers/list/submissions/SubmissionsListHandler.inc.php

```diff
@@ -0,0 +1,159 @@
+
+		$this->_count = isset($args['count']) ? (int) $args['count'] : $this->_count;
+		$this->_getParams = isset($args['getParams']) ? $args['getParams'] : $this->_getParams;
+	}
+
+	/**
+	 * Retrieve the configuration data to be used when initializing this
+	 * handler on the frontend
+	 *
+	 * @return array Configuration data
+	 */
+	public function getConfig() {
+
+		$request = Application::getRequest();
+
+		$config = array();
+
+		if ($this->_lazyLoad) {
+			$config['lazyLoad'] = true;
+		} else {
+			$config['collection'] = $this->getItems();
+		}
+
+		// URL to add a new submission
+		$config['addUrl'] = $request->getDispatcher()->url(
+			$request,
+			ROUTE_PAGE,
+			null,
+			'submission',
+			'wizard'
+		);
+
+		// URL to view info center for a submission
+		$config['infoUrl'] = $request->getDispatcher()->url(
+			$request,
+			ROUTE_COMPONENT,
+			null,
+			'informationCenter.SubmissionInformationCenterHandler',
+			'viewInformationCenter',
+			null,
+			array('submissionId' => '__id__')
+		);
+
+		$config['apiPath'] = $this->_apiPath;
+
+		$config['count'] = $this->_count;
+		$config['page'] = 1;
+
+		$config['getParams'] = $this->_getParams;
+
+		// Load grid localisation files
+		AppLocale::requireComponents(LOCALE_COMPONENT_PKP_GRID);
+		AppLocale::requireComponents(LOCALE_COMPONENT_PKP_SUBMISSION);
+
+		$config['i18n']['title'] = __($this->_title);
+		$config['i18n']['add'] = __('submission.submit.newSubmissionSingle');
+		$config['i18n']['search'] = __('common.search');
+		$config['i18n']['clearSearch'] = __('submission.list.clearSearch');
+		$config['i18n']['itemCount'] = __('submission.list.count');
+		$config['i18n']['itemsOfTotal'] = __('submission.list.itemsOfTotal');
+		$config['i18n']['loadMore'] = __('grid.action.moreItems');
+		$config['i18n']['loading'] = __('common.loading');
+		$config['i18n']['incomplete'] = __('submissions.incomplete');
+		$config['i18n']['delete'] = __('common.delete');
+		$config['i18n']['infoCenter'] = __('submission.list.infoCenter');
+		$config['i18n']['yes'] = __('common.yes');
+		$config['i18n']['no'] = __('common.no');
+		$config['i18n']['deleting'] = __('common.deleting');
+		$config['i18n']['confirmDelete'] = __('submission.list.confirmDelete');
+		$config['i18n']['responseDue'] = __('submission.list.responseDue');
+		$config['i18n']['reviewDue'] = __('submission.list.reviewDue');
+		$config['i18n']['filter'] = __('submission.list.filter');
+		$config['i18n']['filterRemove'] = __('submission.list.filterRemove');
+		$config['i18n']['itemOrdererUp'] = __('submission.list.itemOrdererUp');
+		$config['i18n']['itemOrdererDown'] = __('submission.list.itemOrdererDown');
+		$config['i18n']['reviewsCompleted'] = __('submission.list.reviewsCompleted');
+		$config['i18n']['filesPrepared'] = __('submission.list.filesPrepared');
+		$config['i18n']['discussions'] = __('submission.list.discussions');
+		$config['i18n']['incompleteSubmissionNotice'] = __('submission.list.incompleteSubmissionNotice');
```

Comment:
I think an `array_merge_recursive` call or something would be a lot more readable.